backend/app/main/crud.py

```diff
@@ -0,0 +1,12 @@
+from sqlalchemy.orm import Session
+from . import models, schemas
+
+def get_vendor_by_name(db: Session, name: str):
+    return db.query(models.Vendor).filter(models.Vendor.name == name).first()
+
+def create_vendor(db: Session, vendor: schemas.VendorCreate):
+    db_vendor = models.Vendor(name=vendor.name, address=vendor.address, city=vendor.city)
+    db.add(db_vendor)
+    db.commit()
+    db.refresh(db_vendor)
```

Comment:
can you change `db` to `session`
because it's a session object on which you do .add() .commit(), etc

if I'm not mistaken